enhancements/sdk-kubernetes-1.19.md

```diff
@@ -0,0 +1,249 @@
+with Kubernetes 1.19.
+
+Kubernetes 1.19 has some key changes relevant to Operator development:
+
+- `apiextensions.k8s.io/v1beta1` is deprecated (and scheduled for removal in 1.22) in favor of
+  `apiextensions.k8s.io/v1` ([#90673][90673]) [SIG API Machinery]
+
+[90673]:https://github.com/kubernetes/kubernetes/pull/90673
+
+## Motivation
+
+Supporting Kubernetes 1.19 in operator-sdk is necessary to enable our community to continue using
+the latest and greatest features of Kubernetes.
+
+### Goals
+
+- Operator SDK’s dependencies are upgraded to use Kubernetes 1.19.
+- All e2e tests run during Operator SDK CI are upgraded to test against Kubernetes 1.19.
+- New projects scaffolded with Operator SDK are upgraded to use Kubernetes 1.19 dependencies.
+- New projects scaffolded with Operator SDK should generate v1 CRDs by default.
+
+### Non-Goals
+
+- Feature development and bug fixes that are not directly related to Kubernetes 1.19 support. The
+  rubric here is, if after bumping `go.mod` dependency versions:
+  - Will the test suite fail CI without this change?
+  - Will ignoring this change cause a regression?
+
+  If the answer to these questions is no, then it is out of scope for this enhancement.
+  - Unrelated bug fixes can be done before or after this enhancement is implemented.
+  - Integration of new features introduced in 1.19 will be done in follow-on PRs.
+
+## Proposal
+
+### User Stories
+
+#### Upgrade controller-runtime to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/controller-runtime][gh-controller-runtime]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade controller-tools to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/controller-tools][gh-controller-tools]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Changing the default generated CRD version in controller-gen from v1beta1 to v1.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade kubebuilder to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/kubebuilder][gh-kubebuilder]_
+
+This story consists of:
+
+- Bumping the controller-runtime dependency version in go.mod file scaffolds to the version of
+  controller-runtime that supports Kubernetes 1.19.
+- Bumping the controller-gen dependency version in the Makefile scaffolds to the version of
+  controller-gen that supports Kubernete 1.19.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
```

Comment:
To upgrade KB we need to upgrade also the project: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/go.mod#L10 which is a dependency used for the add-ons

Also, we need to upgrade the version of Webhooks as well and not just CRDs.

Reply:
> To upgrade KB we need to upgrade also the project: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/go.mod#L10 which is a dependency used for the add-ons

I'm not sure this is strictly necessary. Kubebuilder contains all of the scaffolding code for the `addon` pattern. And it is just an import in that scaffold that brings in `kubebuilder-declarative-pattern` into the dependencies of the project.

I'll put a note in here that it may be necessary to bump this version, but this may fall into the out-of-scope bucket given the stated non-goals.

Reply:
Just to clarifies. KB will scaffold a project using `partner=addon` which will use the kb module for it. So, if the module is using an old k8s api version when kb an upper I think it will break the project and then it will not pass in the CI. So, because of this IMO shows required/mandatory too.